### PR TITLE
Changes CI Pipeline and Improves CI Performance

### PR DESCRIPTION
**Pull request by Moejoe90**
image

https://circleci.com/workflow-run/5872b591-81f1-469a-9d77-297694a5802f

All memory issues are solved here also the tests are more rigid. It appears that the lack of memory was a big cause for flaky tests

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -1,79 +1,84 @@
...
+      # - v17-deps-{{ checksum "package.json" }}
+      # - v17-deps-
 
       # Installation
       - run: yarn --frozen-lockfile
       # Linting
       - run: yarn lint
-
-      - run: cp -R ~/.cache/Cypress/ /home/circleci/.cache/
+      # Unit Tests
```
Comment (Moejoe90) [inline .circleci/config.yml]: tests were added to this step since it takes approx 3mins to start a new container and get the cache etc

**Comment by rudolfix** [comment]
hey you could consider adding one more step before e2e tests: pulling all required containers from registry and writing them to cache. then e2e steps will run faster in case of new containers

File: app/test-e2e/register/LightWalletETORegister.spec.e2e.io.ts [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+  const passowrd = "strongpassword";
+
+  it("should register user with light-wallet", () => {
+    cy.clearLocalStorage();
+    registerWithLightWalletETO("moe-wallet-backup-e2e@test.com", passowrd);
+
+    assertEtoDashboard();
+  });
+
+  it("should raise an error that user is already used", () => {
```
Comment (AlexeyTuboltsev) [inline app/test-e2e/register/LightWalletETORegister.spec.e2e.io.ts]: `user is already used` -> 😁

File: app/test-e2e/utils/forms.ts [hunk 3]
```diff
@@ -43,7 +43,8 @@ export const fillField = (key: string, value: string, parent: string = "body") =
   cy.get(parent).within(() => {
     cy.get(formField(key))
       .clear()
-      .type(value);
+      .type(value)
+      .blur();
```
Comment (AlexeyTuboltsev) [inline app/test-e2e/utils/forms.ts]: 👍

File: app/test-e2e/utils/index.ts [hunk 4]
```diff
@@ -82,16 +84,32 @@ export const closeModal = () => {
   cy.get(tid("modal-close-button")).click();
 };
 
-export const getLatestVerifyUserEmailLink = () =>
+export const getLatestVerifyUserEmailLink = (email: string) =>
   cy.request({ url: mockApiUrl + "sendgrid/session/mails", method: "GET" }).then(r => {
-    const activationLink = get(r, "body[0].personalizations[0].substitutions.-activationLink-");
+    const latestEmailByUser = r.body.find(
```
Comment (AlexeyTuboltsev) [inline app/test-e2e/utils/index.ts]: here and below you could use your own fn `getLatestEmailByUser()` from `app/test-e2e/utils/assertions.ts`

**Pull request merged**